d2common/cache.go

```diff
@@ -40,6 +40,18 @@ func (c *Cache) GetBudget() int {
 	return c.budget
 }
 
+func (c *Cache) GetKeys() []string {
+	keys := make([]string, len(c.lookup))
```

Comment:
I think there might be a race conditions here, as even reading, the array could be modified while iterating it. I'm only saying that because I thing the class is supposed to be semi threadsafe